File: sort_algorithm/select_sort.go
package sort_algorithm

import "fmt"

type II = int

type SelectSort []int

func (s SelectSort) SortImpl() {
	for i := 0; i < len(s); i++ {
		index := i
		for j := i + 1; j < len(s); j++ {
			if s[index] > s[j] {
				index = j
			}
		}
		s[i], s[index] = s[index], s[i]
	}

	fmt.Println(s)
}

File: java/sort_algorithm/impl/SelectSort.java
package sort_algorithm.impl;

import sort_algorithm.Sort;

import java.util.Arrays;

public class SelectSort implements Sort {

    @Override
    public void sort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int min = arr[i], index = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (min > arr[j]) {
                    min = arr[j];
                    index = j;
                }
            }
            arr[i] = min;
            int tmp = arr[i];
            arr[index] = tmp;
        }
        System.out.println(Arrays.toString(arr));
    }

}
